Source/ConnectionLostRetryPolicy.swift

```diff
@@ -0,0 +1,106 @@
+/// [technical document](https://developer.apple.com/library/content/qa/qa1941/_index.html).
+public struct ConnectionLostRetryPolicy: RetryPolicy {
+
+    // MARK: - Properties
+
+    /// The total number of times the request is allowed to be retried.
+    public let retryLimit: UInt
+
+    /// The base of the exponential backoff policy (should always be greater than or equal to 2).
+    public let exponentialBackoffBase: UInt
+
+    /// The scale of the exponential backoff.
+    public let exponentialBackoffScale: Double
+
+    /// The idempotent http methods to retry.
+    /// See [RFC 2616 - Section 9.1.2](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html) for more information.
+    public let idempotentMethods: Set<HTTPMethod>
+
+    // MARK: - Initialization
+
+    /// Creates a `ConnectionLostRetryPolicy` instance from the specified parameters.
+    ///
+    /// - Parameters:
+    ///   - retryLimit:              The total number of times the request is allowed to be retried. `2` by default.
+    ///   - exponentialBackoffBase:  The base of the exponential backoff policy. `2` by default.
+    ///   - exponentialBackoffScale: The scale of the exponential backoff. `0.25` by default.
+    ///   - idempotentMethods:       The idempotent http methods to retry. `[.get, .head, .options, .trace]` by default.
+    public init(
+        retryLimit: UInt = 2,
+        exponentialBackoffBase: UInt = 2,
+        exponentialBackoffScale: Double = 0.25,
+        idempotentMethods: Set<HTTPMethod> = [.get, .head, .put, .delete, .options, .trace])
+    {
+        self.retryLimit = retryLimit
+        self.exponentialBackoffBase = exponentialBackoffBase
+        self.exponentialBackoffScale = exponentialBackoffScale
+        self.idempotentMethods = idempotentMethods
+    }
+
+    // MARK: - Retry
+
+    /// Returns whether the request should be retried with a time delay in seconds.
+    ///
+    /// - Parameters:
+    ///   - request: The request to potentially retry.
+    ///   - error:   The error encountered when executing the request.
+    ///
+    /// - Returns: `true` with a time delay if request should be retried, `false` and an ignored time delay otherwise.
+    public func shouldRetry(_ request: Request, with error: Error) -> (shouldRetry: Bool, timeDelay: TimeInterval) {
+        guard
+            request.retryCount < retryLimit,
+            isErrorConnectionLostError(error),
+            let urlRequest = request.request,
+            isRequestIdempotent(urlRequest)
+        else { return (false, 0) }
+
+        // Compute time delay for the exponential backoff (0.5s, 1.0s, 2.0s, 4.0s, ...)
+        let timeDelay = pow(Double(exponentialBackoffBase), Double(request.retryCount)) * exponentialBackoffScale
+
+        return (true, timeDelay)
+    }
+
+    // MARK: - Private - Helper Methods
+
+    private func isRequestIdempotent(_ urlRequest: URLRequest) -> Bool {
```

Comment:
The tricky part with that is that idempotent methods are clearly defined in the [RFC](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html), yet, in practice, `.put` and `.delete` are not always implemented in an idempotent way. Therefore, if we did expose this publicly, you'd have to have a way to modify the default behavior.

> I'm thinking of the way you can add custom content types in AFI as an example.

Otherwise, we'd just have to say that if you need to customize the idempotent methods, you'd need to do that in your own APIs.